test/widget/auth/email_auth_options_page_test.dart

```diff
@@ -651,31 +688,55 @@ void main() {
       expect(invalidPasswordText, findsNothing);
       expect(invalidRepeatPasswordText, findsNothing);
 
-      // input invalid details
+      // Enter invalid details.
       await tester.enterText(emailTextField, 'invalid_email');
       await tester.enterText(passwordTextField, '123');
       await tester.enterText(repeatPasswordTextField, '0');
 
-      // submit invalid form
+      // Attempt to create account.
       await tester.tap(createAccountButton);
 
-      // wait for changes to show
+      // Check that the widget dispatched the expected action.
+      expect(
+          fakeStore.dispatchedActions,
+          contains(UpdateEmailAuthOptionsPage(
+              autovalidate: AutoValidate.onUserInteraction)));
+
+      // Update the state in response to the action.
+      fakeStore.updateState((b) => b
+        ..emailAuthOptionsPage.autovalidate = AutoValidate.onUserInteraction);
+
       await tester.pumpAndSettle();
 
-      // check formfield error messages are shown
+      // Check formfield error messages are shown.
       expect(invalidEmailText, findsOneWidget);
       expect(invalidPasswordText, findsOneWidget);
       expect(invalidRepeatPasswordText, findsOneWidget);
 
-      // input valid details
+      // Enter valid details.
       await tester.enterText(emailTextField, 'valid@email.com');
       await tester.enterText(passwordTextField, 'validPassword');
       await tester.enterText(repeatPasswordTextField, 'validPassword');
 
-      // autovalidation removes error messages
+      // Check that the widget dispatched the expected actions.
+      expect(
+          fakeStore.dispatchedActions,
+          containsAll(<ReduxAction>[
+            UpdateEmailAuthOptionsPage(email: 'valid@email.com'),
+            UpdateEmailAuthOptionsPage(password: 'validPassword'),
+            UpdateEmailAuthOptionsPage(repeatPassword: 'validPassword')
+          ]));
+
+      // Update the state in response to the actions.
+      fakeStore.updateState((b) => b
+        ..emailAuthOptionsPage.email = 'valid@email.com'
+        ..emailAuthOptionsPage.password = 'validPassword'
+        ..emailAuthOptionsPage.repeatPassword = 'validPassword');
+
+      // Let autovalidation remove error messages.
       await tester.pumpAndSettle();
 
-      // check error messages are gone
+      // Check error messages are gone.
```

Comment:
bit confused here, how does the validator know that the fields are valid when we only update the redux state? 
i thought the validator was only linked to the value inside the text fields.

beautiful test btw :) can u please help me wrap my head around this @nickmeinhold, i might just be overthinking 😅 
shouldnt take long 🙏 thank you!